### PR TITLE
Add comments to kernel/platform

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This pull request adds trait comments to the `kernel/platform` files.

This is an attempt to address #1042.

### Testing Strategy

This pull request was tested by travis.

### TODO or Help Wanted

Ideally someone should read over these to make sure what I wrote is correct and potentially add what I didn't include but should have.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: kernel/src/platform/mod.rs [hunk 1]
```diff
@@ -7,26 +7,90 @@ pub mod mpu;
 crate mod systick;
 
 /// Interface for individual boards.
+///
+/// Each board in their `main.rs` file should define a struct which implements
```
Comment (alevy) [inline kernel/src/platform/mod.rs]: Just "Each board", it doesn't have to be in any particular file.

File: kernel/src/platform/mod.rs [hunk 2]
```diff
@@ -7,26 +7,90 @@ pub mod mpu;
...
+/// this trait. This trait is the core for how syscall dispatching is handled.
```
Comment (alevy) [inline kernel/src/platform/mod.rs]: Maybe something more specific: "This trait is responsible for dispatching drivers for each system call number."

File: kernel/src/platform/mod.rs [hunk 3]
```diff
@@ -7,26 +7,90 @@ pub mod mpu;
...
-    /// the Driver methods for that syscall
+    /// the Driver methods for that syscall.
     fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
     where
         F: FnOnce(Option<&dyn Driver>) -> R;
 }
 
 /// Interface for individual MCUs.
+///
+/// Each microcontroller should define a struct and implement this trait.
```
Comment (alevy) [inline kernel/src/platform/mod.rs]: "Defines chip-specific properties of Tock's operation. These include whether and which memory protection mechanism and systick to use, how to dispatch userland applications, and how to handle hardware events.

File: kernel/src/platform/mod.rs [hunk 4]
```diff
@@ -7,26 +7,96 @@ pub mod mpu;
...
 
 /// Interface for individual boards.
+///
+/// Each board should define a struct which implements this trait. This trait is
+/// the core for how syscall dispatching is handled, and the implementation is
+/// responsible for dispatching to drivers for each system call number.
+///
+/// ## Example
+///
+/// ```ignore
```
Comment (alistair23) [inline kernel/src/platform/mod.rs]:
```rust doesn't work? Is it worth fixing the error to avoid the documentation bit rotting?

**Pull request merged**